dashboard/src/app/(authenticated)/(diagram)/wfSpec/[...props]/components/SearchVariableDialog.tsx

```diff
@@ -0,0 +1,147 @@
+        wfSpecRevision: spec.id!.revision,
+        tenantId,
+        limit,
+        bookmark: pageParam,
+        varName: variable.name,
+        value: convertToVariableValue({ type: variable.type.toLowerCase(), value: variableValueDebounced }),
+      })
+    },
+  })
+  return (
+    <Dialog>
+      <DialogTrigger asChild>
+        <Button>Search By Variable</Button>
+      </DialogTrigger>
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>Search WfRuns by Variable</DialogTitle>
+          <DialogDescription>
+            Search for WfRuns by variable name and value. The search is case-sensitive.
+          </DialogDescription>
+        </DialogHeader>
+
+        <Select
+          onValueChange={value => setVariable(variables.find(v => v.name == value) ?? variable)}
+          value={variable.name}
+        >
+          <SelectTrigger>
+            <SelectValue placeholder="Select Variable" />
+          </SelectTrigger>
+          <SelectContent>
+            {variables.map(varDef => (
+              <SelectItem key={varDef.name} value={varDef.name}>
+                {varDef.name}
+              </SelectItem>
+            ))}
+          </SelectContent>
+        </Select>
+        <Input
+          placeholder="Variable Value..."
+          onChange={e => {
+            setVariableValue(e.target.value)
+          }}
+        />
+
+        {isPending ? (
+          <div className="flex items-center justify-center">
+            <RefreshCwIcon className="h-8 w-8 animate-spin text-blue-500" />
+          </div>
+        ) : data?.pages[0].results.length ? (
+          data?.pages.map((page, i) => (
+            <Fragment key={i}>
+              {page.results.map(variableId => (
+                <div key={variableId.name}>
+                  <Link
+                    className="py-2 text-blue-500 hover:underline"
+                    href={`/wfRun/${variableId.wfRunId?.id}?threadRunNumber=${variableId.threadRunNumber}`}
+                  >
+                    {variableId.wfRunId?.id}
+                  </Link>
+                </div>
+              ))}
+            </Fragment>
+          ))
+        ) : (
+          <p className="text-center">
+            No data. Try another value for <span className="font-bold">{variable.name}</span>
+          </p>
+        )}
+        <SearchFooter
+          currentLimit={limit}
+          setLimit={setLimit}
+          hasNextPage={hasNextPage}
+          fetchNextPage={fetchNextPage}
+        />
+      </DialogContent>
+    </Dialog>
+  )
+}
+
+function convertToVariableValue({ type, value }: { type: string; value: string }): VariableValue {
+  switch (type) {
+    case 'str':
+      return { str: value }
+    case 'int':
+      return { int: parseInt(value) }
+    case 'bool':
+      return { bool: value.toLowerCase() === 'true' }
+    case 'double':
+      return { double: parseFloat(value) }
+    default:
+      return { str: value }
+  }
```

Comment:
This should converted by the form instead, but this implementation is sufficient we should use a different approach if we plan to support `JSON_OBJ` and `JSON_ARR`